geomstats/geometry/poincare_ball.py

```diff
@@ -227,32 +229,73 @@ def mobius_add(self, point_a, point_b):
         return mobius_add
 
     @geomstats.vectorization.decorator(['else', 'vector', 'vector'])
-    def dist(self, point_a, point_b):
-        """Compute the geodesic distance between two points.
+    def dist(self, points_a, points_b):
+        """Compute the geodesic distance between points.
 
         Parameters
         ----------
-        point_a : array-like, shape=[..., dim]
-            First point in hyperbolic space.
-        point_b : array-like, shape=[..., dim]
-            Second point in hyperbolic space.
+        points_a : array-like, shape=[n_samples_a, dim]
+            Set of points in hyperbolic space.
+        points_b : array-like, shape=[n_samples_b, dim]
+            Second set of points in hyperbolic space.
 
         Returns
         -------
-        dist : array-like, shape=[...,]
-            Geodesic distance between the two points.
+        dist : array-like,
+            shape=[n_samples_a, dim] or [n_samples_a, n_samples_b, dim]
+            Geodesic distance between the two set of points.
+            If n_samples_a == n_samples_b then dist is the element-wise
```

Comment:
This explanation could go into the description of the method, and not of the output.